### PR TITLE
Add github workflow

### PR DESCRIPTION
**Pull request by klemmari1**
- Update README.
- Add editorconfig.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'python' ]
```
Comment (nenonja) [inline .github/workflows/codeql-analysis.yml:31]: javascript could be added?

File: .github/workflows/staging.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
+name: Build & Staging
+on:
+  push:
+#    branches:
```
Comment (nenonja) [inline .github/workflows/staging.yml:4]: why commented?

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '33 22 * * 1'
```
Comment (nenonja) [inline .github/workflows/codeql-analysis.yml:21]: seems a bit odd scheduled time? https://crontab.guru/#33_22_*_*_1

**Review by nenonja — COMMENTED** [review]
Quickly browsed through the code and added some comments for the parts I understood :)

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (klemmari1) [reply .github/workflows/codeql-analysis.yml]: Yep, I'll add :+1:

File: .github/workflows/staging.yml (hunk 2 above)
Comment (klemmari1) [reply .github/workflows/staging.yml]: This was commented to test out that deployment to staging works before we push to the `main` branch. I'll uncomment it now that it is finally working :D

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (klemmari1) [reply .github/workflows/codeql-analysis.yml:21]: This is an example setup I'm using from https://github.com/City-of-Helsinki/kuva-demo-project. I think it is just to run it once a week, I don't know if the time matters.

Comment (nenonja) [reply .github/workflows/codeql-analysis.yml:21]: This could be more convenient: https://crontab.guru/every-week

**Review by nenonja — APPROVED** [review]
Looks good now for me

**Pull request merged**